gateway.py

```diff
@@ -697,6 +697,20 @@ def delete_listing(http: urllib3.PoolManager, auth_token, listing_id):
     if not creds:
         return make_unauthorized_response()
 
+    lookup_result = execute_data_get(
+        http, f"/get_listing?listingId={listing_id}")
+
+    if lookup_result.status == 200:
```

Comment:
if this lookup fails then the function will continue. i know i probably coded it this way in patch listings too but we should add
```
else:
    return make_internal_server_error()
```